From c19fdeb723a3259f0d9bedfe0b8e8f4fe2fbdd29 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 18 Jan 2024 14:28:33 +0100
Subject: [PATCH 0492/1695] dts: stm32: sync stm32mp15 with 3.19.0-stm32mp-dev

Synchronize STM32MP15 DTS/DTSI files with 3.19.0-stm32mp-dev branch
commit 2cc216af311d ("plat-stm32mp2: enable async notif with GIC PPI 15").

- Add missing reset controller handle in STM32MP15 SoC DTSI nodes.
- Add missing counter instance in STM32MP15 timer3 node.
- Add missing PMIC regulators delay properties.
- Add missing BSEC clock reference.
- Enable digital bypass on HSE clock.
- Fix alignment issues in DTSI files.
- Disable many devices.
- Define board_id BSEC NVMEM node and serial0..2 aliases from
  stm32mp15xx-dkx.dtsi instead of each stm32mp157?-dk*.dtsi files.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0cb0221d4e10a6cf34860d7bf2fd6f92f6814e5f dts: stm32: sync stm32mp15 with 3.19.0-stm32mp-dev]
Change-Id: Ib2b1c920ae73dc9565ab714acb2a5d8777f53543
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/352661
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/dts/stm32mp151.dtsi            | 19 +++-
 core/arch/arm/dts/stm32mp157a-dk1.dts        |  8 +-
 core/arch/arm/dts/stm32mp157a-ed1.dts        |  5 ++
 core/arch/arm/dts/stm32mp157c-dk2.dts        |  7 --
 core/arch/arm/dts/stm32mp157c-ed1.dts        | 28 ++++--
 core/arch/arm/dts/stm32mp157c-ev1.dts        | 30 ++++---
 core/arch/arm/dts/stm32mp157d-ed1.dts        |  5 ++
 core/arch/arm/dts/stm32mp157f-ed1.dts        |  5 ++
 core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi |  4 +-
 core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi |  2 +-
 core/arch/arm/dts/stm32mp15xx-dkx.dtsi       | 92 ++++++++++++--------
 11 files changed, 134 insertions(+), 71 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp151.dtsi b/core/arch/arm/dts/stm32mp151.dtsi
index be5e9d22c..dbfdb5b2f 100644
--- a/core/arch/arm/dts/stm32mp151.dtsi
+++ b/core/arch/arm/dts/stm32mp151.dtsi
@@ -458,6 +458,7 @@
 			dmas = <&dmamux1 43 0x400 0x15>,
 			       <&dmamux1 44 0x400 0x11>;
 			dma-names = "rx", "tx";
+			resets = <&rcc USART2_R>;
 			status = "disabled";
 		};
 
@@ -470,6 +471,7 @@
 			dmas = <&dmamux1 45 0x400 0x15>,
 			       <&dmamux1 46 0x400 0x11>;
 			dma-names = "rx", "tx";
+			resets = <&rcc USART3_R>;
 			status = "disabled";
 		};
 
@@ -482,6 +484,7 @@
 			dmas = <&dmamux1 63 0x400 0x15>,
 			       <&dmamux1 64 0x400 0x11>;
 			dma-names = "rx", "tx";
+			resets = <&rcc UART4_R>;
 			status = "disabled";
 		};
 
@@ -494,6 +497,7 @@
 			dmas = <&dmamux1 65 0x400 0x15>,
 			       <&dmamux1 66 0x400 0x11>;
 			dma-names = "rx", "tx";
+			resets = <&rcc UART5_R>;
 			status = "disabled";
 		};
 
@@ -603,6 +607,7 @@
 			dmas = <&dmamux1 79 0x400 0x15>,
 			       <&dmamux1 80 0x400 0x11>;
 			dma-names = "rx", "tx";
+			resets = <&rcc UART7_R>;
 			status = "disabled";
 		};
 
@@ -615,6 +620,7 @@
 			dmas = <&dmamux1 81 0x400 0x15>,
 			       <&dmamux1 82 0x400 0x11>;
 			dma-names = "rx", "tx";
+			resets = <&rcc UART8_R>;
 			status = "disabled";
 		};
 
@@ -699,6 +705,7 @@
 			dmas = <&dmamux1 71 0x400 0x15>,
 			       <&dmamux1 72 0x400 0x11>;
 			dma-names = "rx", "tx";
+			resets = <&rcc USART6_R>;
 			status = "disabled";
 		};
 
@@ -755,6 +762,11 @@
 			dma-names = "ch1", "up", "trig", "com";
 			status = "disabled";
 
+			counter {
+				compatible = "st,stm32-timer-counter";
+				status = "disabled";
+			};
+
 			pwm {
 				compatible = "st,stm32-pwm";
 				#pwm-cells = <3>;
@@ -1529,6 +1541,7 @@
 		bsec: efuse@5c005000 {
 			compatible = "st,stm32mp15-bsec";
 			reg = <0x5c005000 0x400>;
+			clocks = <&rcc BSEC>;
 			#address-cells = <1>;
 			#size-cells = <1>;
 
@@ -1820,7 +1833,7 @@
 				reg = <0x5c009000 0x400>;
 				interrupt-names = "event", "error";
 				interrupts = <GIC_SPI 135 IRQ_TYPE_LEVEL_HIGH>,
-						<GIC_SPI 136 IRQ_TYPE_LEVEL_HIGH>;
+					     <GIC_SPI 136 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc I2C6_K>;
 				resets = <&rcc I2C6_R>;
 				#address-cells = <1>;
@@ -1852,8 +1865,8 @@
 			reg = <0x10000000 0x40000>,
 			      <0x30000000 0x40000>,
 			      <0x38000000 0x10000>;
-			resets = <&rcc MCU_R>;
-			st,syscfg-holdboot = <&rcc 0x10C 0x1>;
+			resets = <&rcc MCU_R>, <&rcc MCU_HOLD_BOOT_R>;
+			reset-names = "mcu_rst", "hold_boot";
 			st,syscfg-tz = <&rcc 0x000 0x1>;
 			st,syscfg-pdds = <&pwr_mcu 0x0 0x1>;
 			st,syscfg-rsc-tbl = <&tamp 0x144 0xFFFFFFFF>;
diff --git a/core/arch/arm/dts/stm32mp157a-dk1.dts b/core/arch/arm/dts/stm32mp157a-dk1.dts
index 9a6b7d1cb..c12c1c8a1 100644
--- a/core/arch/arm/dts/stm32mp157a-dk1.dts
+++ b/core/arch/arm/dts/stm32mp157a-dk1.dts
@@ -8,6 +8,7 @@
 
 #include <dt-bindings/soc/stm32mp15-etzpc.h>
 #include "stm32mp157.dtsi"
+#include "stm32mp15xa.dtsi"
 #include "stm32mp15-pinctrl.dtsi"
 #include "stm32mp15xxac-pinctrl.dtsi"
 #include "stm32mp15xx-dkx.dtsi"
@@ -18,9 +19,6 @@
 
 	aliases {
 		ethernet0 = &ethernet0;
-		serial0 = &uart4;
-		serial1 = &usart3;
-		serial2 = &uart7;
 	};
 
 	chosen {
@@ -29,10 +27,6 @@
 };
 
 &bsec {
-	board_id: board_id@ec {
-		reg = <0xec 0x4>;
-		st,non-secure-otp;
-	};
 	huk_otp: huk-otp@f0 {
 		reg = <0xf0 0x10>;
 	};
diff --git a/core/arch/arm/dts/stm32mp157a-ed1.dts b/core/arch/arm/dts/stm32mp157a-ed1.dts
index 769040755..b56b2d3c2 100644
--- a/core/arch/arm/dts/stm32mp157a-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157a-ed1.dts
@@ -214,6 +214,11 @@
 	status = "okay";
 };
 
+&pwr_regulators {
+	vdd-supply = <&vdd>;
+	vdd_3v3_usbfs-supply = <&vdd_usb>;
+};
+
 &rcc {
 	st,clksrc = <
 		CLK_MPU_PLL1P
diff --git a/core/arch/arm/dts/stm32mp157c-dk2.dts b/core/arch/arm/dts/stm32mp157c-dk2.dts
index b5f858058..a3ea2424f 100644
--- a/core/arch/arm/dts/stm32mp157c-dk2.dts
+++ b/core/arch/arm/dts/stm32mp157c-dk2.dts
@@ -18,9 +18,6 @@
 
 	aliases {
 		ethernet0 = &ethernet0;
-		serial0 = &uart4;
-		serial1 = &usart3;
-		serial2 = &uart7;
 		serial3 = &usart2;
 	};
 
@@ -30,10 +27,6 @@
 };
 
 &bsec {
-	board_id: board_id@ec {
-		reg = <0xec 0x4>;
-		st,non-secure-otp;
-	};
 	huk_otp: huk-otp@f0 {
 		reg = <0xf0 0x10>;
 	};
diff --git a/core/arch/arm/dts/stm32mp157c-ed1.dts b/core/arch/arm/dts/stm32mp157c-ed1.dts
index d19aba921..7b65f0d33 100644
--- a/core/arch/arm/dts/stm32mp157c-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157c-ed1.dts
@@ -114,7 +114,7 @@
 		st,adc-channels = <0 1 6>;
 		/* 16.5 ck_cycles sampling time */
 		st,min-sample-time-nsecs = <400>;
-		status = "okay";
+		status = "disabled";
 	};
 };
 
@@ -123,11 +123,16 @@
 		reg = <0xec 0x4>;
 		st,non-secure-otp;
 	};
+
 	huk_otp: huk-otp@f0 {
 		reg = <0xf0 0x10>;
 	};
 };
 
+&clk_hse {
+	st,digbypass;
+};
+
 &crc1 {
 	status = "disabled";
 };
@@ -138,10 +143,10 @@
 	vref-supply = <&vdda>;
 	status = "disabled";
 	dac1: dac@1 {
-		status = "okay";
+		status = "disabled";
 	};
 	dac2: dac@2 {
-		status = "okay";
+		status = "disabled";
 	};
 };
 
@@ -269,12 +274,12 @@
 				regulator-max-microvolt = <750000>;
 				regulator-always-on;
 				regulator-over-current-protection;
+				st,regulator-sink-source;
 			};
 
 			vdd_usb: ldo4 {
 				regulator-name = "vdd_usb";
 				interrupts = <IT_CURLIM_LDO4 0>;
-				regulator-always-on;
 			};
 
 			vdd_sd: ldo5 {
@@ -295,6 +300,7 @@
 			vref_ddr: vref_ddr {
 				regulator-name = "vref_ddr";
 				regulator-always-on;
+				regulator-over-current-protection;
 			};
 
 			bst_out: boost {
@@ -305,13 +311,13 @@
 			vbus_otg: pwr_sw1 {
 				regulator-name = "vbus_otg";
 				interrupts = <IT_OCP_OTG 0>;
-			 };
+			};
 
-			 vbus_sw: pwr_sw2 {
+			vbus_sw: pwr_sw2 {
 				regulator-name = "vbus_sw";
 				interrupts = <IT_OCP_SWOUT 0>;
 				regulator-active-discharge = <1>;
-			 };
+			};
 		};
 
 		onkey {
@@ -529,6 +535,14 @@
 	/delete-property/dmas;
 	/delete-property/dma-names;
 	timer@5 {
+		status = "disabled";
+	};
+};
+
+&timers15 {
+	status = "okay";
+
+	counter {
 		status = "okay";
 	};
 };
diff --git a/core/arch/arm/dts/stm32mp157c-ev1.dts b/core/arch/arm/dts/stm32mp157c-ev1.dts
index a938ae7a6..82eb20c67 100644
--- a/core/arch/arm/dts/stm32mp157c-ev1.dts
+++ b/core/arch/arm/dts/stm32mp157c-ev1.dts
@@ -67,7 +67,7 @@
 		compatible = "gpio-backlight";
 		gpios = <&gpiod 13 GPIO_ACTIVE_LOW>;
 		default-on;
-		status = "okay";
+		status = "disabled";
 	};
 };
 
@@ -121,7 +121,7 @@
 		reset-gpios = <&gpiof 15 GPIO_ACTIVE_LOW>;
 		backlight = <&panel_backlight>;
 		power-supply = <&v3v3>;
-		status = "okay";
+		status = "disabled";
 
 		port {
 			dsi_panel_in: endpoint {
@@ -157,7 +157,7 @@
 	status = "disabled";
 
 	nand-controller@4,0 {
-		status = "okay";
+		status = "disabled";
 
 		nand@0 {
 			reg = <0>;
@@ -185,7 +185,7 @@
 		powerdown-gpios = <&stmfx_pinctrl 18 (GPIO_ACTIVE_HIGH | GPIO_PUSH_PULL)>;
 		reset-gpios = <&stmfx_pinctrl 19 (GPIO_ACTIVE_LOW | GPIO_PUSH_PULL)>;
 		rotation = <180>;
-		status = "okay";
+		status = "disabled";
 
 		port {
 			ov5640_0: endpoint {
@@ -222,6 +222,16 @@
 	};
 };
 
+&i2c4 {
+	pmic: stpmic@33 {
+		regulators {
+			v1v8: ldo6 {
+				regulator-enable-ramp-delay = <300000>;
+			};
+		};
+	};
+};
+
 &i2c5 {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&i2c5_pins_a>;
@@ -304,10 +314,10 @@
 		pinctrl-0 = <&pwm2_pins_a>;
 		pinctrl-1 = <&pwm2_sleep_pins_a>;
 		pinctrl-names = "default", "sleep";
-		status = "okay";
+		status = "disabled";
 	};
 	timer@1 {
-		status = "okay";
+		status = "disabled";
 	};
 };
 
@@ -319,10 +329,10 @@
 		pinctrl-0 = <&pwm8_pins_a>;
 		pinctrl-1 = <&pwm8_sleep_pins_a>;
 		pinctrl-names = "default", "sleep";
-		status = "okay";
+		status = "disabled";
 	};
 	timer@7 {
-		status = "okay";
+		status = "disabled";
 	};
 };
 
@@ -334,10 +344,10 @@
 		pinctrl-0 = <&pwm12_pins_a>;
 		pinctrl-1 = <&pwm12_sleep_pins_a>;
 		pinctrl-names = "default", "sleep";
-		status = "okay";
+		status = "disabled";
 	};
 	timer@11 {
-		status = "okay";
+		status = "disabled";
 	};
 };
 
diff --git a/core/arch/arm/dts/stm32mp157d-ed1.dts b/core/arch/arm/dts/stm32mp157d-ed1.dts
index 198a606db..6a100adb4 100644
--- a/core/arch/arm/dts/stm32mp157d-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157d-ed1.dts
@@ -213,6 +213,11 @@
 	status = "okay";
 };
 
+&pwr_regulators {
+	vdd-supply = <&vdd>;
+	vdd_3v3_usbfs-supply = <&vdd_usb>;
+};
+
 &rcc {
 	st,clksrc = <
 		CLK_MPU_PLL1P
diff --git a/core/arch/arm/dts/stm32mp157f-ed1.dts b/core/arch/arm/dts/stm32mp157f-ed1.dts
index 6ccf6fdea..10b9b9b14 100644
--- a/core/arch/arm/dts/stm32mp157f-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157f-ed1.dts
@@ -214,6 +214,11 @@
 	};
 };
 
+&pwr_regulators {
+	vdd-supply = <&vdd>;
+	vdd_3v3_usbfs-supply = <&vdd_usb>;
+};
+
 &rcc {
 	compatible = "st,stm32mp1-rcc-secure";
 
diff --git a/core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi b/core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi
index b4fc6da5b..5172c09e4 100644
--- a/core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi
@@ -312,9 +312,9 @@
 
 			vdda: ldo1 {
 				regulator-name = "vdda";
-				regulator-always-on;
 				regulator-min-microvolt = <2900000>;
 				regulator-max-microvolt = <2900000>;
+				regulator-always-on;
 				interrupts = <IT_CURLIM_LDO1 0>;
 			};
 
@@ -342,8 +342,8 @@
 				regulator-name = "vdd_sd";
 				regulator-min-microvolt = <2900000>;
 				regulator-max-microvolt = <2900000>;
-				interrupts = <IT_CURLIM_LDO5 0>;
 				regulator-boot-on;
+				interrupts = <IT_CURLIM_LDO5 0>;
 			};
 
 			v1v8: ldo6 {
diff --git a/core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi b/core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi
index 312f3870a..de0d86e23 100644
--- a/core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi
@@ -175,8 +175,8 @@
 				regulator-name = "v1v8";
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
-				interrupts = <IT_CURLIM_LDO6 0>;
 				regulator-enable-ramp-delay = <300000>;
+				interrupts = <IT_CURLIM_LDO6 0>;
 			};
 
 			vref_ddr: vref_ddr {
diff --git a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
index 9c5a64acc..9ad2de50a 100644
--- a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
@@ -11,6 +11,12 @@
 #include <dt-bindings/tamper/st,stm32-tamp.h>
 
 / {
+	aliases {
+		serial0 = &uart4;
+		serial1 = &usart3;
+		serial2 = &uart7;
+	};
+
 	memory@c0000000 {
 		device_type = "memory";
 		reg = <0xc0000000 0x20000000>;
@@ -70,7 +76,6 @@
 			gpios = <&gpiod 11 GPIO_ACTIVE_HIGH>;
 			linux,default-trigger = "heartbeat";
 			default-state = "off";
-			status = "disabled";
 		};
 	};
 
@@ -111,13 +116,20 @@
 		st,min-sample-time-nsecs = <5000>;
 		/* AIN connector, USB Type-C CC1 & CC2 */
 		st,adc-channels = <0 1 6 13 18 19>;
-		status = "okay";
+		status = "disabled";
 	};
 	adc2: adc@100 {
 		/* AIN connector, USB Type-C CC1 & CC2 */
 		st,adc-channels = <0 1 2 6 18 19>;
 		st,min-sample-time-nsecs = <5000>;
-		status = "okay";
+		status = "disabled";
+	};
+};
+
+&bsec {
+	board_id: board_id@ec {
+		reg = <0xec 0x4>;
+		st,non-secure-otp;
 	};
 };
 
@@ -182,7 +194,7 @@
 		interrupts = <1 IRQ_TYPE_EDGE_FALLING>;
 		interrupt-parent = <&gpiog>;
 		#sound-dai-cells = <0>;
-		status = "okay";
+		status = "disabled";
 
 		ports {
 			#address-cells = <1>;
@@ -215,7 +227,7 @@
 		reset-gpios = <&gpiog 9 GPIO_ACTIVE_LOW>;
 		clocks = <&sai2a>;
 		clock-names = "MCLK";
-		status = "okay";
+		status = "disabled";
 
 		cs42l51_port: port {
 			#address-cells = <1>;
@@ -238,6 +250,10 @@
 	};
 };
 
+&clk_hse {
+	st,digbypass;
+};
+
 &i2c4 {
 	compatible = "st,stm32mp15-i2c-non-secure";
 	pinctrl-names = "default", "sleep";
@@ -258,7 +274,7 @@
 		interrupt-parent = <&gpioi>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&stusb1600_pins_a>;
-		status = "okay";
+		status = "disabled";
 		vdd-supply = <&vin>;
 
 		connector {
@@ -355,16 +371,14 @@
 
 			vtt_ddr: ldo3 {
 				regulator-name = "vtt_ddr";
-				regulator-min-microvolt = <500000>;
-				regulator-max-microvolt = <750000>;
 				regulator-always-on;
 				regulator-over-current-protection;
+				st,regulator-sink-source;
 			};
 
 			vdd_usb: ldo4 {
 				regulator-name = "vdd_usb";
 				interrupts = <IT_CURLIM_LDO4 0>;
-				regulator-always-on;
 			};
 
 			vdda: ldo5 {
@@ -388,21 +402,21 @@
 				regulator-always-on;
 			};
 
-			 bst_out: boost {
+			bst_out: boost {
 				regulator-name = "bst_out";
 				interrupts = <IT_OCP_BOOST 0>;
-			 };
+			};
 
 			vbus_otg: pwr_sw1 {
 				regulator-name = "vbus_otg";
 				interrupts = <IT_OCP_OTG 0>;
-			 };
+			};
 
-			 vbus_sw: pwr_sw2 {
+			vbus_sw: pwr_sw2 {
 				regulator-name = "vbus_sw";
 				interrupts = <IT_OCP_SWOUT 0>;
 				regulator-active-discharge = <1>;
-			 };
+			};
 		};
 
 		onkey {
@@ -410,7 +424,7 @@
 			interrupts = <IT_PONKEY_F 0>, <IT_PONKEY_R 0>;
 			interrupt-names = "onkey-falling", "onkey-rising";
 			power-off-time-sec = <10>;
-			status = "okay";
+			status = "disabled";
 		};
 
 		watchdog {
@@ -459,11 +473,6 @@
 	status = "okay";
 };
 
-&iwdg2 {
-	timeout-sec = <32>;
-	status = "disabled";
-};
-
 &ltdc {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&ltdc_pins_a>;
@@ -485,7 +494,7 @@
 	mbox-names = "vq0", "vq1", "shutdown", "detach";
 	interrupt-parent = <&exti>;
 	interrupts = <68 1>;
-	status = "okay";
+	status = "disabled";
 };
 
 &pwr_regulators {
@@ -635,7 +644,7 @@
 		dma-names = "tx";
 		clocks = <&rcc SAI2_K>;
 		clock-names = "sai_ck";
-		status = "okay";
+		status = "disabled";
 
 		sai2a_port: port {
 			sai2a_endpoint: endpoint {
@@ -653,7 +662,7 @@
 		st,sync = <&sai2a 2>;
 		clocks = <&rcc SAI2_K>, <&sai2a>;
 		clock-names = "sai_ck", "MCLK";
-		status = "okay";
+		status = "disabled";
 
 		sai2b_port: port {
 			sai2b_endpoint: endpoint {
@@ -707,10 +716,10 @@
 		pinctrl-0 = <&pwm1_pins_a>;
 		pinctrl-1 = <&pwm1_sleep_pins_a>;
 		pinctrl-names = "default", "sleep";
-		status = "okay";
+		status = "disabled";
 	};
 	timer@0 {
-		status = "okay";
+		status = "disabled";
 	};
 };
 
@@ -722,10 +731,10 @@
 		pinctrl-0 = <&pwm3_pins_a>;
 		pinctrl-1 = <&pwm3_sleep_pins_a>;
 		pinctrl-names = "default", "sleep";
-		status = "okay";
+		status = "disabled";
 	};
 	timer@2 {
-		status = "okay";
+		status = "disabled";
 	};
 };
 
@@ -737,10 +746,10 @@
 		pinctrl-0 = <&pwm4_pins_a &pwm4_pins_b>;
 		pinctrl-1 = <&pwm4_sleep_pins_a &pwm4_sleep_pins_b>;
 		pinctrl-names = "default", "sleep";
-		status = "okay";
+		status = "disabled";
 	};
 	timer@3 {
-		status = "okay";
+		status = "disabled";
 	};
 };
 
@@ -752,10 +761,10 @@
 		pinctrl-0 = <&pwm5_pins_a>;
 		pinctrl-1 = <&pwm5_sleep_pins_a>;
 		pinctrl-names = "default", "sleep";
-		status = "okay";
+		status = "disabled";
 	};
 	timer@4 {
-		status = "okay";
+		status = "disabled";
 	};
 };
 
@@ -764,7 +773,7 @@
 	/delete-property/dma-names;
 	status = "disabled";
 	timer@5 {
-		status = "okay";
+		status = "disabled";
 	};
 };
 
@@ -776,9 +785,17 @@
 		pinctrl-0 = <&pwm12_pins_a>;
 		pinctrl-1 = <&pwm12_sleep_pins_a>;
 		pinctrl-names = "default", "sleep";
-		status = "okay";
+		status = "disabled";
 	};
 	timer@11 {
+		status = "disabled";
+	};
+};
+
+&timers15 {
+	status = "okay";
+
+	counter {
 		status = "okay";
 	};
 };
@@ -858,9 +875,16 @@
 	st,no-lsfs-sc;
 };
 
+&usart3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&usart3_pins_c>;
+	uart-has-rtscts;
+	status = "disabled";
+};
+
 &vrefbuf {
 	regulator-min-microvolt = <2500000>;
 	regulator-max-microvolt = <2500000>;
 	vdda-supply = <&vdd>;
-	status = "okay";
+	status = "disabled";
 };
-- 
2.47.2

